src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -1202,5 +1279,53 @@ public static void GenerateNupkgMetadataFile(string nupkgPath, string installPat
                     }
                 });
         }
+
+        private static T[] CancellableEnumaration<T>(IEnumerable<T> enumerable, ILogger log, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // .ToArray is necessary because previously it returns Array
+                // If enumaration happen several times then same I/O calls repeatedly called on same input, I/O calls are more expensive then memory.
+                return CancellableYieldEnumaration(enumerable, log, cancellationToken).ToArray();
```

Comment:
Use ToList instead of ToArray, it copies less. ToArray forces the underlying array to be the exact same size as the number of results, whereas ToList allows the underlying array to be larger than the number of elements.

Reply:
Sure, replaced with List.